### PR TITLE
Add a benchmark for bridging String(contentsOfFile:...)

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

File: benchmark/single-source/ObjectiveCBridging.swift [hunk 1]
```diff
@@ -709,3 +712,35 @@ public func run_ObjectiveCBridgeFromNSDateComponents(_ N: Int) {
   }
   #endif
 }
+
+var stringFromFile:String? = nil
+public func setup_stringFromFile() {
+  #if _runtime(_ObjC)
+  let url:URL
+  if #available(OSX 10.12, iOS 10.0, *) {
```
Comment (Catfish-Man) [inline benchmark/single-source/ObjectiveCBridging.swift]: bleh

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

File: benchmark/single-source/ObjectiveCBridging.swift [hunk 2]
```diff
@@ -709,3 +712,35 @@ public func run_ObjectiveCBridgeFromNSDateComponents(_ N: Int) {
...
+    url = FileManager.default.temporaryDirectory.appendingPathComponent(
+      "sphinx.txt"
+    )
+  } else {
+    url = URL(fileURLWithPath: "/tmp/sphinx.txt")
```
Comment (milseman) [inline benchmark/single-source/ObjectiveCBridging.swift]: Does Foundation have a mktemp equivalent?

**Review by milseman — APPROVED** [review]
If the intent is for this to be all-ASCII, add ASCII to the name.

File: benchmark/single-source/ObjectiveCBridging.swift (hunk 2 above)
Comment (Catfish-Man) [reply benchmark/single-source/ObjectiveCBridging.swift]: The call in the other else branch is the best way, there's a clunkier version I could use on old OSs if it's a problem

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by palimondo** [comment]
Can somebody please explain the intention behind this benchmark?

All this does is run `.utf8.count` on a string that is **in memory**, but gets there in an extremely convoluted way during lazy initialization in the `setUpFunction` by writing a string to temporary file and loading it back in.

The `setUpFunction` get's executed only once before gathering multiple samples, not before every sample!

If you meant to test the performance of `String(contentsOf:, encoding:)`, than you need to store the `url` in a global variable and move the [last line from `setup_ASCIIStringFromFile`](https://github.com/apple/swift/pull/24326/files#diff-615270fa3eab8efa0288636dddd1f7dfR734) to the loop in `run_ASCIIStringFromFile`.

CC @Catfish-Man, @milseman, @eeckstein

**Comment by palimondo** [comment]
> All this does is run .utf8.count on a string that is in memory…

I take that back! Looking at #24386, I cannot believe the bridging can lead to this unbelievably convoluted and underperforming implementation behind the scenes. That is all kinds of wrong in my opinion. Improvements in #24386 lead to zero runtime of the `ObjectiveCBridgeASCIIStringFromFile`, so I believe this should not have been a benchmark in the first place, just a bug. I think the question still stands:

> Can somebody please explain the intention behind this benchmark?

**Comment by milseman** [comment]
@Catfish-Man can probably provide more specific rationale, but there's a big issue with `String(contentsOfFile:)` even for all-ASCII files. CF does not believe the ASCII string is nul-terminated, so it will report back as being neither contiguous UTF-8 nor contiguous UTF-16. Instead, it is as though it were some arbitrary subclass of NSString and accessing the contents has to go through message sends fetching UTF-16 code units. I don't know if the URL-based one has similar issues (I suspect it would if loading from a file). This is likely part of the machinery in CF related to loading files.

**Pull request merged**